### ISSUE TITLE
[BUG] Race-condition for media with both queued and nonQueued conversions

### ISSUE THREAD

**Issue by naabster** [spatie/laravel-medialibrary#2226]
possibly connected to #2213 and #1304

Situation: I have a model with multiple conversions, one of them is nonQueued:

    public function registerMediaConversions(Media $media = null): void
    {
        $this->addMediaConversion('card')->nonQueued()->fit('crop', 736, 464);
        $this->addMediaConversion('thumb')->fit('crop', 368, 232)->sharpen(10);
        $this->addMediaConversion('medium')->width(1600)->height(1600);
    }

What we are seeing: Randomly, new uploaded media will have either 'thumb' and 'medium' or only 'card' in generated_conversions column. If you look at the file system, all three conversions are actually there, but the database is not up to date. This happens on a single queue worker locally as well as on Vapor based deployments. 

I think the Problem is a typical race condition: 

```
class FileManipulator
{
    public function createDerivedFiles(
        Media $media,
        array $onlyConversionNames = [],
        bool $onlyMissing = false
    ): void {
        if (!$this->canConvertMedia($media)) {
            return;
        }

        [$queuedConversions, $conversions] = ConversionCollection::createForMedia($media)
            ->filter(function (Conversion $conversion) use ($onlyConversionNames) {
                if (count($onlyConversionNames) === 0) {
                    return true;
                }

                return in_array($conversion->getName(), $onlyConversionNames);
            })
            ->filter(fn (Conversion $conversion) => $conversion->shouldBePerformedOn($media->collection_name))
            ->partition(fn (Conversion $conversion) => $conversion->shouldBeQueued());

        $this
            ->dispatchQueuedConversions($media, $queuedConversions, $onlyMissing)
            ->performConversions($conversions, $media, $onlyMissing);
    }
```

--> queued conversions are dispatched
--> nonQueued conversions are handled

It might happen, that the queued conversions are actually so fast, that they are colliding with the nonQueued conversion.
The `markAsConversionGenerated()` method on the media model reads the generated conversions, adds the new ones and saves to DB. 

Possible Solutions:
* We would need some sort of locking here (https://laravel.com/docs/master/queries#pessimistic-locking).
* Another approach could be to first handle nonQueued conversions and dispatch queued conversions after finishing them.

**Comment by DaniTulp** [comment]
I experience the same behaviour, I think it's especially frequent if the files being converted have large resolutions. I use [Picsum](https://picsum.photos/3000) for testing if I have a picture that's 3000x3000 it pretty much always overwrites the nonQueued conversions with the queued conversions in the database. I also feel like adding more nonQueued conversions makes it worse even with lower resolution files.

**Comment by lostincode** [comment]
I've been running into this as well. It's really hard to reproduce and make sense that it could be a race condition. In my case, I'm using media library pro and have a non queued `preview` conversion and it randomly disappears from the database but the conversion is created.

**Comment by freekmurze** [comment]
This has been fixed by #2235
